src/diskImage.cpp

```diff
@@ -12,14 +17,63 @@ void DiskImage::setPath(const std::filesystem::path& path)
     if (path.extension() != ".iso")
         throw std::runtime_error("Disk Image must have ISO extension");
 
+    // Verify checksum only if the image is known.
+    if (isKnownImage(path)) {
+        if (!hasVerifiedChecksum(path))
+            throw std::runtime_error("Disk Image checksum isn't valid");
+    }
+
     m_path = path;
 }
 
-bool DiskImage::isKnownImage()
+bool DiskImage::isKnownImage(const std::filesystem::path& path)
 {
     for (const auto& image : m_knownImageFilename)
-        if (m_path.filename() == image)
+        if (path.filename().string() == image) {
+            LOG_TRACE("Disk image is recognized");
             return true;
+        }
+
+    LOG_TRACE("Disk image is unknown. Maybe you're using a custom image or "
+              "changed the default name?");
+    return false;
+}
+
+bool DiskImage::hasVerifiedChecksum(const std::filesystem::path& path)
+{
+    using namespace CryptoPP;
+    using namespace std;
+
+    LOG_TRACE("Verifying disk image checksum... This may take a while");
+
+    map<string, string> hash_map = {
+        { "rhel-8.8-x86_64-dvd.iso",
+            "517abcc67ee3b7212f57e180f5d30be3e8269e7a99e127a3399b7935c7e00a0"
+            "9" },
+        { "OracleLinux-R8-U8-x86_64-dvd.iso",
+            "cae39116245ff7c3c86d5305d9c11430ce5c4e512987563435ac59c37a082d7"
+            "e" },
+        { "Rocky-8.8-x86_64-dvd1.iso",
+            "7b8bdfe189cf24ae5c2d6a88f7a0b5f3012d23f9332c47943d538b4bc03a370"
+            "4" },
+        { "AlmaLinux-8.8-x86_64-dvd.iso",
+            "635b30b967b509a32a1a3d81401db9861922acb396d065922b39405a43a04a3"
+            "1" },
+    };
+
+    SHA256 hash;
+    string isoHash = hash_map.find(path.filename().string())->second;
+    string output;
+    FileSource(path.string().c_str(), true,
+        new HashFilter(hash, new HexEncoder(new StringSink(output))), true);
```

Comment:
You don't want to use `new`. And also this generates a memory leak since I could not find any `delete` on the code.

Use smart pointers instead: `std::unique_ptr`. Using a smart pointer will guarantee that the memory will be free when the object goes out of scope.

Reply:
I did it according to the documentation ([FileSource](https://www.cryptopp.com/wiki/FileSource#:~:text=A%20FileSource%20allows%20you%20to,sink%20object%20is%20a%20FileSink) and [HashFilter](https://www.cryptopp.com/wiki/HashFilter#:~:text=HashFilter%20uses%20the%20specified%20hash,a%20pointer%20to%20a%20BufferedTransformation%20))

They are deleted after used.